src/users/src/users-service/handlers.go

```diff
@@ -270,44 +262,49 @@ func UserVerifyAndUpdatePhone(w http.ResponseWriter, r *http.Request){
 		if err != nil {
 			fmt.Println("Got error calling PhoneNumberValidate:")
 			fmt.Println(err.Error())
+			http.Error(w, err.Error(), 500)
+			return
 		} else {
-				fmt.Println(res)
-				mobilePhoneCode := int(*res.NumberValidateResponse.PhoneTypeCode)
-				if (mobilePhoneCode != 0) {
-					panic("Received a phone number that isn't capable of receiving SMS. Cannot create endpoint")
-				} else {
-					var userAge string = strconv.Itoa(user.Age)
-					var userAttributes = make(map[string][]*string)
-					userAttributes["Username"] = []*string{&user.Username}
-					userAttributes["FirstName"] = []*string{&user.FirstName}
-					userAttributes["LastName"] = []*string{&user.LastName}
-					userAttributes["Gender"] = []*string{&user.Gender}
-					userAttributes["Age"] = []*string{&userAge}
-
-					endpointRequest := &pinpoint.EndpointRequest{
-						Address: &userdetails.PhoneNumber,
-						ChannelType: aws.String("SMS"),
-						OptOut: aws.String("ALL"),
-						Location: &pinpoint.EndpointLocation {
-							PostalCode: res.NumberValidateResponse.ZipCode,
-							City: res.NumberValidateResponse.City,
-							Country: res.NumberValidateResponse.CountryCodeIso2,
-						},
-						Demographic: &pinpoint.EndpointDemographic {
-							Timezone: res.NumberValidateResponse.Timezone,
-						},
-						User: &pinpoint.EndpointUser{
-							UserAttributes: userAttributes,
-							UserId: &userdetails.Userid,
-						},
-					}
-					var endpointId string = userdetails.PhoneNumber[1:]
-					updateEndpointInput := &pinpoint.UpdateEndpointInput{
-						ApplicationId: &pinpoint_app_id,
-						EndpointId: &endpointId,
-						EndpointRequest: endpointRequest,
-					}
-					CreateEndpointAndSendConfirmation(updateEndpointInput, userdetails.PhoneNumber)
+			fmt.Println(res)
+			mobilePhoneCode := int(*res.NumberValidateResponse.PhoneTypeCode)
+			if (mobilePhoneCode != 0) {
+				var errMessage string = "The phone number provided is not a MOBILE phone number. The number is not capable of receiving SMS. Cannot create SMS endpoint for this number. Try entering a mobile phone number."
+				panic(errMessage)
+				http.Error(w, errMessage, 500)
```

Comment:
500 doesn't seem like the right error code here. The caller provided a bad phone number so something like a 422 or 400 would be more appropriate.

Reply:
Okay, I will change it to 422